### PR TITLE
[wasm] Switch default modules to es6

### PR DESCRIPTION
**Pull request by pavelsavara**
## Motivation, benefits 
- TL;DR: 
    - consistent in NodeJS and in browsers
    - module isolation
    - well defined exports/imports
    - dependencies resolution
    - dependencies detected during parsing, not during run
    - async dynamic imports awaitable
- [Why Modules](https://eloquentjavascript.net/10_modules.html)
- Supported everywhere for many years, everywhere where WASM is able to run.

## Challenges, downsides
- CommonJS module could be [wrapped inside simple ES6 script](https://redfin.engineering/node-modules-at-war-why-commonjs-and-es-modules-cant-get-along-9617135eeca1). **But not the other way around.**
    - ES6 modules could be re-packaged by standard tools like webpack, rollup back to CJS or UMD.
    - libraries using another ES6 modules have to be also ES6 module (unless they re-pack with tools above).
- Node.js 12 doesn't support it, but is end-of-life by 2022-04-30
- After this PR will would still have way how to produce CommonJS, but [is it worth supporting it](https://github.com/dotnet/runtime/issues/63337) ?

## Changes in this PR
- switched dotnet.js to be ES6 module by making `<WasmEnableES6>` default true
- updated all samples
- updated functional tests
- updated debugger tests
- updated test-main to
- `EXPORT_ES6=1` should not be in the `emcc-link.rsp` because it could not be removed by workload re-link when `<WasmEnableES6> == false`.

### CODE REVIEW

File: src/mono/sample/mbr/DeltaHelper/DeltaHelper.csproj [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>$(NetCoreAppCurrent)</TargetFramework>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (kg) [inline src/mono/sample/mbr/DeltaHelper/DeltaHelper.csproj:3]: why? considering we're moving towards 7 this seems odd

**Comment by radical** [comment]
I can review once this is out of draft.

File: src/mono/sample/mbr/DeltaHelper/DeltaHelper.csproj (hunk 1 above)
Comment (pavelsavara) [reply src/mono/sample/mbr/DeltaHelper/DeltaHelper.csproj]: Thanks, that was a mistake.

**Comment by maraf** [comment]
WebPack sample doesn't work right now.

The reason is that webpack replaces `import.meta.url` with a compile time constant (= path on disk, eg. `file:///C:/...`). The runtime is than trying to load all assets as local files. This webpack behavior is an open issue https://github.com/webpack/webpack/issues/14445.

**Pull request closed without merging**